.github/workflows/docker-build.yml

```
@@ -0,0 +1,20 @@
+name: Docker Image Build CI
+
+on:
+  push:
+    branches: [ "dev", "main" ]
+  pull_request:
+    branches: [ "dev", "main" ]
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Build the Docker image
+      run: docker build . --file Dockerfile --tag my-image-name:$(date +%s) --build-arg="github_token=$GITHUB_TOKEN"
```

Minor nitpick since this is just a CI validation workflow:

```suggestion
      run: docker build . --file Dockerfile --tag linode/cli:$(date +%s) --build-arg="github_token=$GITHUB_TOKEN"
```